transformer_engine/pytorch/module/linear.py

```diff
@@ -609,18 +634,31 @@ def __init__(
 
         self.sequence_parallel = (self.tp_size > 1) and sequence_parallel
 
-        self.weight_tensor = torch.empty(
+        temp_weight = torch.empty(
             self.out_features, self.in_features,
             device=device, dtype=params_dtype)
 
+        # TODO(ksivaman): This functionality works with FP8 outside TE.
         initialize_affine_weight_gpu(
-            self.weight_tensor,
+            temp_weight,
             init_method,
             get_rng_state_tracker,
             partition_dim=1 if self.parallel_mode == "row" else 0,
             stride=1,
         )
 
+        if self.primary_weights_in_fp8:
+            self.fp8_init()
+            self.fp8_meta["update_amax_and_scale_fwd"] = True
+
+            self.weight_tensor = Float8Tensor.to_float8(
+                temp_weight,
+                fp8_meta=self.fp8_meta,
```

Comment:
With `fp8_init`, there's no option to specify the "recipe" but the weight tensor would store the reference to `fp8_meta` created during module initialization, which I guess will always point to the default recipe config. Right, @ksivaman? (@timmoon10 for reference)